internal/configs/resource.go

```diff
@@ -813,59 +929,29 @@ func (m *ProviderConfigRefMapping) decodeStaticAlias(eval *StaticEvaluator, forE
 		})
 	}
 
-	// TODO/Oleksandr: Is static identifier valid if we evaluate for_each for alias?
-	instances, instancesDiags := eval.Evaluate(forEach, aliasStaticID)
+	instances, instancesDiags := eval.Evaluate(instanceExpr, aliasStaticID)
```

Comment:
This should probably use evalchecks (or where ever it ends up landing) to properly handle decoding the each/count data.  I'd be happy with a TODO for later.

Reply:
Thanks! Left a todo for later, when we update feature branch with [the Andrew's PR](https://github.com/opentofu/opentofu/pull/1911)